### PR TITLE
Add ability to customize GDB port when auto-starting gdb server

### LINKED ISSUE #569: Feature request: add option to connect to a different OpenOCD port
### Description

In the Debugger tab I can choose the GDB, Telnet and Tcl ports, this is fine and it configures the ports when OpenOCD is invoked.
However I cannot choose a different port GDB number but the one that is specified in the option above.
The use case is related to multicore devices with different (and consecutive ports) for the different cores, e.g. core 1 port 3333, core 2 port 3334, and so on.
At the moment I can only connect to the core 1, i.e. port 3333 and if I specify 3334 as GDB port, it will connect to port 3334 but this will still be the 1st core, with 2nd core having port 3334 and so on. 
If I deselect the flag "Start OpenOCD locally" then for sure I can specify the exact port I want GDB to connect to but with the hassle to start OpenOCD manually.

It would be nice to have the possibility to specify the actual GDB port.
I tried to use -ex "target extended-remote localhost:3334" as "other options" but it does not work.

### Versions

* [plug-in version] 6.3.2
* [Eclipse version] 2022-06 (4.24.0)
* [Java version]
* [operating system] Ubuntu 22.04
* [toolchain version] NA

### PR DESCRIPTION
**Pull request by jonahgraham**
When starting a gdbserver (like openocd) connected to a multi-core SoC the gdb server may open multiple ports. This change allows user to configure which port GDB connect to rather than forcing GDB to connec to the first port.

TODO list:

- [ ] Get approval for this idea in #569
- [x] Save the new checkbox state properly
- [ ] Review wording of the checkbox and tooltip (at the moment it is a bit wordy and maybe not clear enough)
- [ ] Apply this same change to all the other TabDebugger.java files

Fixes #569

### CODE REVIEW

**Comment by ilg-ul** [comment]
> wording of the checkbox 

yes, this is a hard one. :-(

**Comment by ilg-ul** [comment]
> wording of the checkbox

How about "In sync with the GDB server"?

**Comment by jonahgraham** [comment]
Replaced by solution #571
